Baby Monitor/Source Files/Modules/CameraPreview/CameraPreviewViewModel.swift

```diff
@@ -82,4 +96,22 @@ final class CameraPreviewViewModel: BaseViewModel {
             print("connection status: connecting")
         }
     }
+
+    private func rxSetupMicrophoneEvents() {
+        microphoneHoldEvent?
+            .subscribe(onNext: { [weak self] _ in
+                guard let self = self else { return }
+                if self.isMicrophoneAccessGranted {
+                    self.webSocketWebRtcService.get().startAudioTransmitting()
+                } else {
+                    self.noMicrophoneAccessPublisher.onNext(())
+                }
+            }).disposed(by: bag)
+        microphoneReleaseEvent?
+            .subscribe(onNext: { [weak self] _ in
+                guard let self = self,
+                    self.isMicrophoneAccessGranted else { return }
```

Comment:
Could be also lifted to a `.filter` - this way it's more declarative. Just a suggestion though